### ISSUE TITLE
Saturn Style Router

### ISSUE THREAD

**Issue by AngelMunoz** [AngelMunoz/Bix#1]
**Is your feature request related to a problem? Please describe.**

The default router we have is fairly straight forward but it might become verbose and annoying to write for larger applications, it also doesn't help to figure out the structure of a web application

**Describe the solution you'd like**

The way Bix uses routes is similar to the EndpointRouter from [Saturn](https://github.com/SaturnFramework/Saturn), we define a list of routes and then we match on them via URLPattern, Ideally we should be able to create a similar router we might not have pipelines and plugs at the beginning but at least routers and controllers should be present

```fsharp
let todoController: RouteDefinition list = 
  controller {
    // operation: HttpContext -> Promise<BixResponse>
    find (ctx -> BixResponse) // get /todos
    findOne (id -> ctx -> BixResponse) // get /todos/:id
    update (ctx -> BixResponse) // put /todos
    updateOne (id -> ctx -> BixResponse) // put /todos/:id
    create (ctx -> BixResponse) // post /todos
    delete (ctx -> BixResponse) // delete /todos
  }

let routes: RouteDefinition list = 
  router {
    // verbOperation -> string -> HttpHandler
    get url HttpHandler
    post url HttpHandler
    put url HttpHandler
    patch url HttpHandler
    delete url HttpHandler
    custom verb url HttpHandler
    forward url handler // ideal
    // alternative in case the above proves to be complicated
    controller "/todos" todoController 
  }
```

**Describe alternatives you've considered**
A plain Giraffe style of routing

**Additional context**

**Comment by AngelMunoz** [comment]
Some updates
![image](https://user-images.githubusercontent.com/8684875/179670047-e98ff19e-5468-4dd6-bbb6-cb40c817bac7.png)

**Comment by AngelMunoz** [comment]
this was merged in #2
